### PR TITLE
[gimkl/2.11.5] OpenFOAM-Extend 3.1 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 17 out of 18 (18 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/806cb050bfb980f4bb86 for a full test report.

**Comment by wpoely86** [comment]
@boegel have you seen this one before: `FAIL (unhandled exception: AttributeError) Python-2.7.10-gimkl-2.11.5.eb ` ?

**Comment by boegel** [comment]
@wpoely86: hmm, that may be a bug somewhere in an easyblock that is not handled well by the test report mechanism... do you have the output of the `eb` run that submitted this test report?

Or can you try building that Python easyconfig manually yourself?

**Comment by wpoely86** [comment]
The logs say 18 of 18 successful. It's weird, I don't see anything wrong.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 17 out of 18 (18 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/52cb12e4c04bec6a9bdb for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 25 out of 27 (18 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/3f8eca03c4cb6810d12a for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/43dd50a6ccc5a5efb36b for a full test report.

File: easybuild/easyconfigs/t/Tk/Tk-8.6.4-gimkl-2.11.5-no-X11.eb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+homepage = 'http://www.tcl.tk/'
+description = """Tk is an open source, cross-platform widget toolchain that provides a library of basic elements for building
+ a graphical user interface (GUI) in many different programming languages."""
+
+toolchain = {'name': 'gimkl', 'version': '2.11.5'}
+
+source_urls = ["http://prdownloads.sourceforge.net/tcl"]
+sources = ['%(namelower)s%(version)s-src.tar.gz']
+
```
Comment (boegel) [inline easybuild/easyconfigs/t/Tk/Tk-8.6.4-gimkl-2.11.5-no-X11.eb]: please include the patch from #2102

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/45933b70e5f78e6617e4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a48069ee30764be3e1ab for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/3b9d41a9ddc45ebb3492 for a full test report.

File: easybuild/easyconfigs/t/Tk/Tk-8.6.4-gimkl-2.11.5-no-X11.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/t/Tk/Tk-8.6.4-gimkl-2.11.5-no-X11.eb]: ok now

File: easybuild/easyconfigs/p/Python/Python-2.7.10-gimkl-2.11.5.eb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+sources = [SOURCE_TGZ]
+
+# python needs bzip2 to build the bz2 package
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.8'),
+    ('libreadline', '6.3'),
+    ('ncurses', '5.9'),
+    ('SQLite', '3.8.10.2'),
+    ('Tk', '8.6.4', '-no-X11'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.10-gimkl-2.11.5.eb]: GMP should be included here too, required for pycrypto extension

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.10-gimkl-2.11.5.eb]: hmm, apparently this hasn't been done consistently in other Python 2.7.10 easyconfigs... so let's sync this up in a follow up PR

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/dc239a10d9590e4d1736 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**